components/ai_chat/resources/page/components/input_box/index.tsx

```diff
@@ -8,57 +8,53 @@ import * as React from 'react'
 import classnames from 'classnames'
 import { getLocale } from '$web-common/locale'
 import Icon from '@brave/leo/react/icon'
-import styles from './style.module.scss'
 
-interface InputBoxProps {
-  onInputChange?: (e: React.ChangeEvent<HTMLTextAreaElement>) => void
-  onSubmit?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
-  onSummaryClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
-  onKeyDown?: (e: React.KeyboardEvent<HTMLTextAreaElement>) => void
-  value: string
-  hasSeenAgreement: boolean
-  disabled: boolean
-  onHandleAgreeClick: Function
-}
+import styles from './style.module.scss'
+import DataContext from '../../state/context'
+import getPageHandlerInstance, { APIError } from '../../api/page_handler'
 
 const MAX_INPUT_CHAR = 2000
 const CHAR_LIMIT_THRESHOLD = MAX_INPUT_CHAR * 0.80
 
-function InputBox (props: InputBoxProps) {
-  const [inputText, setInputText] = React.useState(props.value)
+function InputBox () {
+  const [inputText, setInputText] = React.useState('')
+  const { currentError, isGenerating, hasSeenAgreement, handleAgreeClick } = React.useContext(DataContext)
 
   const isCharLimitExceeded = inputText.length >= MAX_INPUT_CHAR
   const isCharLimitApproaching = inputText.length >= CHAR_LIMIT_THRESHOLD
 
-  if (!props.hasSeenAgreement) {
-    const handleAgreeClick = () => {
-      props.onHandleAgreeClick()
-    }
+  const apiHasError = !!currentError && (currentError !== APIError.None)
+  const isInputDisabled = apiHasError || isGenerating || isCharLimitExceeded
```

Comment:
This seems important to share on the context so that suggested questions are disabled or hidden?